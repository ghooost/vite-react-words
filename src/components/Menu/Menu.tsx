import { NavLink } from "react-router-dom";
import { Base } from "@router/contants";

import styles from "./styles.module.css";

const getClassName = ({
  isActive,
  isPending,
}: {
  isActive: boolean;
  isPending: boolean;
}) =>
  [
    styles.item,
    isActive ? styles.active : "",
    isPending ? styles.pending : "",
  ].join(" ");

export const Menu = function () {
  return (
    <nav className={styles.root}>
      <NavLink to={Base} end className={getClassName}>
        Quiz
      </NavLink>
      <NavLink to={`${Base}setup`} className={getClassName}>
        Setup
      </NavLink>
    </nav>
  );
};
