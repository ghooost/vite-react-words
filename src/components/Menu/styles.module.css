.root {
  flex: 1 1 auto;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  border-bottom: 1px solid black;
}

.item {
  padding: 1em;
  display: block;
  text-decoration: none;
  color: black;
}

.active {
  color: white;
  background-color: black;
}

.pending {
  color: white;
  background-color: black;
  opacity: 0.5;
}
