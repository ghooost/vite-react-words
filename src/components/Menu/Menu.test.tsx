import { afterEach, vi, describe, expect, it } from "vitest";
import { Menu } from "./Menu";
import { renderToString } from "react-dom/server";
import { StaticRouter } from "react-router-dom/server";

describe("Menu", () => {
  afterEach(() => {
    vi.restoreAllMocks();
  });

  it("Menu /", () => {
    const out = renderToString(
      <StaticRouter location="/">
        <Menu />
      </StaticRouter>
    );
    expect(out).toMatchSnapshot();
  });

  it("Menu /setup", () => {
    const out = renderToString(
      <StaticRouter location="/setup">
        <Menu />
      </StaticRouter>
    );
    expect(out).toMatchSnapshot();
  });

  it("Menu /setup/something", () => {
    const out = renderToString(
      <StaticRouter location="/setup/something">
        <Menu />
      </StaticRouter>
    );
    expect(out).toMatchSnapshot();
  });

  it("Menu /about", () => {
    const out = renderToString(
      <StaticRouter location="/about">
        <Menu />
      </StaticRouter>
    );
    expect(out).toMatchSnapshot();
  });
});
