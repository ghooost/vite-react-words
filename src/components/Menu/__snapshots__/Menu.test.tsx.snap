// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Menu > Menu / 1`] = `"<nav class=\\"_root_ff9f9e\\"><a aria-current=\\"page\\" class=\\"_item_ff9f9e _active_ff9f9e \\" href=\\"/\\">Quiz</a><a class=\\"_item_ff9f9e  \\" href=\\"/setup\\">Setup</a><a class=\\"_item_ff9f9e  \\" href=\\"/about\\">About</a></nav>"`;

exports[`Menu > Menu /about 1`] = `"<nav class=\\"_root_ff9f9e\\"><a class=\\"_item_ff9f9e  \\" href=\\"/\\">Quiz</a><a class=\\"_item_ff9f9e  \\" href=\\"/setup\\">Setup</a><a aria-current=\\"page\\" class=\\"_item_ff9f9e _active_ff9f9e \\" href=\\"/about\\">About</a></nav>"`;

exports[`Menu > Menu /setup 1`] = `"<nav class=\\"_root_ff9f9e\\"><a class=\\"_item_ff9f9e  \\" href=\\"/\\">Quiz</a><a aria-current=\\"page\\" class=\\"_item_ff9f9e _active_ff9f9e \\" href=\\"/setup\\">Setup</a><a class=\\"_item_ff9f9e  \\" href=\\"/about\\">About</a></nav>"`;

exports[`Menu > Menu /setup/something 1`] = `"<nav class=\\"_root_ff9f9e\\"><a class=\\"_item_ff9f9e  \\" href=\\"/\\">Quiz</a><a aria-current=\\"page\\" class=\\"_item_ff9f9e _active_ff9f9e \\" href=\\"/setup\\">Setup</a><a class=\\"_item_ff9f9e  \\" href=\\"/about\\">About</a></nav>"`;
